rules/REQUEST-931-APPLICATION-ATTACK-RFI.conf

```diff
@@ -105,12 +105,12 @@ SecRule TX:EXECUTING_PARANOIA_LEVEL "@lt 2" "id:931014,phase:2,pass,nolog,skipAf
 # -= Paranoia Level 2 =- (apply only when tx.executing_paranoia_level is sufficiently high: 2 or higher)
 #
 
-SecRule ARGS "@rx ^(?i:file|ftps?|https?)://([^/]*).*$" \
+SecRule ARGS "@rx ^(?i:file|ftps?|https?)://(?:[^@]+@)?([^/]*).*$" \
```

Comment:
The modification is only need so you can capture the domain cleanly, correct? I can't see any bypasses this would allow immediately, however, if the attacker tries a bit harder, the capture would still become "unclean". Example: with `http://example.com%2f@1@2@evil.com/` the capture would contain `1@2@evil.com`. I'm not sure, whether it's worth the effort (and potentially risk) to add this to the expression.

Reply:
uhm I found a possible bypass with our new regex, by using `#` here:
`https://evil.com%2f%23@localhost`

```shell
curl -s -H 'Host: example.com' 'http://localhost:8000/?url=http://evil.com%2f%23@example.com' | grep title
   <title>Evil.Com - We get it...Daily.</title>
```

:(

Reply:
I'm going to update this PR with your regex and create a test on our sandbox so we can easily test it